techmatters/flex-plugins
# CHI 2889 Implement Status column in Teams View with sorting

## Description
This PR is for implementing a Status column (move from Agent column) in the Teams View with sorting functionality with `sortStatusColumn`. 
- Note that a useEffect was created to calculate the dynamic display of time (based on interval or 1s or 1 min)
- Implemented truncating and tooltip for activities with long (12+) characters
- For sorting, created `convertDurationToSeconds` to handle special Twilio's display (Convert duration string in the format "59s", "59:59", "23h" or "23h 59min", "59d" or "5d 3h"  to seconds)
- Minor refactors to reuse styled-components
- I moved the teams view work to the `components` directory.

### Checklist
- [x] Corresponding issue has been [opened](https://tech-matters.atlassian.net/browse/CHI-2889)

### Other Related Issues

None

### Verification steps
- Check sorting logic (Sorted available workers first, offline workers last, then by activity index, then within each activity by duration, with longest to shortest)
- Test and check styling details against [mocks](https://www.figma.com/design/fMuQjby2d6FEqGNMW1qXIH/Teams-View?node-id=581-14507&t=FVke1VhOUG6YDRbN-0) , notably truncating and tooltip behavior
![Screenshot 2024-08-13 at 2 40 55 PM](https://github.com/user-attachments/assets/3a162df5-df71-4f83-aa62-25bc3d6edf09)


### AFTER YOU MERGE

1. Cut a release tag using the Github workflow. Wait for it to complete and notify in the #aselo-deploys Slack channel.
2. Comment on the ticket with the release tag version AND any additional instructions required to configure an environment to test the changes.
3. Only then move the ticket into the QA column in JIRA

You are responsible for ensuring the above steps are completed. If you move a ticket into QA without advising what version to test, the QA team will assume the latest tag has the changes. If it does not, the following confusion is on you! :-P
